server/protocols/jmap/src/main/java/org/apache/james/jmap/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -0,0 +1,168 @@
+
+    @Inject
+    @VisibleForTesting
+    SetMailboxesUpdateProcessor(MailboxUtils<Id> mailboxUtils, MailboxManager mailboxManager) {
+        this.mailboxUtils = mailboxUtils;
+        this.mailboxManager = mailboxManager;
+    }
+
+    @Override
+    public SetMailboxesResponse process(SetMailboxesRequest request, MailboxSession mailboxSession) {
+        SetMailboxesResponse.Builder responseBuilder = SetMailboxesResponse.builder();
+        request.getUpdate()
+            .entrySet()
+            .stream()
+            .forEach(update -> handleUpdate(update, responseBuilder, mailboxSession));
+        return responseBuilder.build();
+    }
+
+    private void handleUpdate(Entry<String, MailboxUpdateRequest> update, Builder responseBuilder, MailboxSession mailboxSession) {
+        String mailboxId = update.getKey();
+        MailboxUpdateRequest updateRequest = update.getValue();
+        try {
+            ensureMailboxName(updateRequest, mailboxSession);
+            Mailbox mailbox = getMailbox(mailboxId, updateRequest, mailboxSession);
+            ensureParent(mailbox, updateRequest, mailboxSession);
+
+            MailboxPath originMailboxPath = mailboxUtils.getMailboxPath(mailbox, mailboxSession);
+            MailboxPath destinationMailboxPath = computeNewMailboxPath(mailbox, originMailboxPath, updateRequest, mailboxSession);
+            if (!originMailboxPath.equals(destinationMailboxPath)) {
+                mailboxManager.renameMailbox(originMailboxPath, destinationMailboxPath, mailboxSession);
+            }
+            responseBuilder.updated(update.getKey());
+        } catch (MailboxNameException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description(e.getMessage())
+                    .build());
+        } catch (MailboxNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The mailbox '%s' was not found", mailboxId))
+                    .build());
+        } catch (MailboxParentNotFoundException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("notFound")
+                    .description(String.format("The parent mailbox '%s' was not found.", e.getParentId()))
+                    .build());
+        } catch (MailboxHasChildException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("invalidArguments")
+                    .description("Cannot update a parent mailbox.")
+                    .build());
+        } catch (MailboxException e) {
+            responseBuilder.notUpdated(mailboxId, SetError.builder()
+                    .type("anErrorOccurred")
+                    .description("An error occurred when updating the mailbox")
+                    .build());
+        }
```

Comment:
I think this method is a bit too long. Would it make sens to split it in two ?
One for your computation, one other for error handling ?